cpp/src/Slice/Parser.h

```diff
@@ -702,16 +705,17 @@ class Operation : public virtual Contained, public virtual Container
     size_t returnBitSequenceSize() const;
 
     void destroy() override;
-    TypePtr returnType() const;
-    bool returnIsTagged() const;
-    int returnTag() const;
+    TypePtr returnType() const; //TODO remove this once the compilers have been updated to use return-tuples.
+    bool returnIsTagged() const; //TODO remove this once the compilers have been updated to use return-tuples.
+    int returnTag() const; //TODO remove this once the compilers have been updated to use return-tuples.
     Mode mode() const;
     Mode sendMode() const;
     bool hasMarshaledResult() const;
     MemberPtr createParameter(const std::string&, const TypePtr&, bool, bool, int);
     MemberList parameters() const;
```

Comment:
Why do we have parameters, inParameters and outParameters?

I assume it should be just params and all the other functions are temporary / to be removed?

Reply:
Yes you're correct, I'll mark the other methods with my "TODO remove..." comment. When everything is done, there should only be 'parameters' left of these 3.